### DISCUSSION TITLE
Open menu on left click at X,Y coordinates, only when state is set to true.

### DISCUSSION THREAD

**Discussion by rahul3103** [radix-ui/primitives#2723] [Help] [answered]
So I have a chart where I can't wrap each bar with the Trigger Component. I can use the Dropdown Menu component like this.

```
<DropdownMenu open={isOpen} onOpenChange={updateDialogState}>
    <DropdownMenuTrigger
          style={{
            top: clickPosition.Y,
            left: clickPosition.X,
            position: "absolute",
          }}
        />
    <DropdownMenuContent className="w-28 h-56"></DropdownMenuContent>
</DropdownMenu>
```

However, assigning a location like this defeats the purpose, as the menu content cannot be adjusted when there is no room at the bottom or right. I want a menu to open with a left click, but it should also adapt its position if there is no room on the right, top, left, or bottom. Is there a way I can open the menu without it being cut?

**Comment by benoitgrelard** [comment]
Do you have a sandbox showing what you are trying to achieve?

**Reply by rahul3103** [reply]
Open menu on right click
I am achieving this by : https://codesandbox.io/p/devbox/fxkynl
Problem with this approach is collision detection doesn't work.

**Reply by benoitgrelard** [reply, marked as answer]
Closing as duplicate of #2726
